analyses/cell-type-wilms-tumor-06/scripts/06a_build-geneposition.R

```diff
@@ -1,25 +1,87 @@
 # This script downloads and adapt the genome position file for use with infercnv
 
+# load library
+library(tidyverse)
+library(stringr)
+
+
 # The base path for the OpenScPCA repository, found by its (hidden) .git directory
 repository_base <- rprojroot::find_root(rprojroot::is_git_root)
 # The path to this module
 module_base <- file.path(repository_base, "analyses", "cell-type-wilms-tumor-06")
 
-# load library
-library(tidyverse)
-# Define the gene order file
-  gene_order_file <- file.path(module_base, "results","references", 'gencode_v19_gene_pos.txt')
+# Define the file paths
+gene_order_file <- file.path(module_base, "results", "references", "gencode_v38_gene_pos.txt")
+arm_order_file <- file.path(module_base, "results", "references", "hg38_cytoBand.txt")
+gene_arm_order_file <- file.path(module_base, "results", "references", "gencode_v38_gene_pos_arm.txt")
+
+if (!file.exists(gene_arm_order_file)) {
+  
+  # Download the updated gene order file for GENCODE v38 (Ensembl 104)
+  if (!file.exists(gene_order_file)) {
+    download.file(url = "https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/gencode.v38.annotation.gtf.gz",
+                  destfile = gene_order_file)
+  }
+  
+  # Read the GTF file and extract gene positions
+  gtf_data <- read.table(gzfile(gene_order_file), header = FALSE, sep = "\t", comment.char = "#", stringsAsFactors = FALSE)
+  
+  # Filter for gene entries only
+  gene_order <- gtf_data %>%
```

Comment:
Before entering this code, rename the data frame and then use those names directly - this is what you do for the cytoband file, and it would help here too.